### PR TITLE
add unit tests for timer and verify updates

### PR DESCRIPTION
**Pull request by ariovistus**

### CODE REVIEW

File: wpilib/tests/test_timer.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    assert timer.hasPeriodPassed(3.0) == True
+    assert timer.startTime == pytest.approx(8000.0, 0.01)
+    assert timer.get() == pytest.approx(0.1, 0.01)
+
+
+    sim_hooks.time = 14.1
+    assert timer.get() == pytest.approx(6.1, 0.01)
+
+    # eh?
+    assert timer.hasPeriodPassed(3.0) == True
```
Comment (ariovistus) [inline wpilib/tests/test_timer.py]: I suppose if you aren't calling this frequently, you deserve to get non periodic behavior? still..

File: wpilib/wpilib/timer.py [hunk 2]
```diff
@@ -34,19 +34,15 @@ def getFPGATimestamp():
...
     def getMatchTime():
         """Return the approximate match time.
-        The FMS does not currently send the official match time to the robots.
-        This returns the time since the enable signal sent from the Driver
-        Station.
-        At the beginning of autonomous, the time is reset to 0.0 seconds.
-        At the beginning of teleop, the time is reset to +15.0 seconds.
-        If the robot is disabled, this returns 0.0 seconds.
-
-        .. warning::
```
Comment (virtuald) [inline wpilib/wpilib/timer.py]: The warning needs this special formatting.

Comment (ariovistus) [reply wpilib/wpilib/timer.py]: better?

Comment (virtuald) [reply wpilib/wpilib/timer.py]: Missed a colon, and it requires indenting at the level of the w.

File: wpilib/wpilib/timer.py [hunk 3]
```diff
@@ -34,19 +34,17 @@ def getFPGATimestamp():
...
-        Station.
-        At the beginning of autonomous, the time is reset to 0.0 seconds.
-        At the beginning of teleop, the time is reset to +15.0 seconds.
-        If the robot is disabled, this returns 0.0 seconds.
-
+        The FMS does not currently send the official match time to the robots,
+        but does send an approximate match time. The value will count down
+        the time remaining in the current period (auto or teleop). 
+        The Practice Match function of the DS approximates the behavior seen
+        on the field.
```
Comment (auscompgeek) [inline wpilib/wpilib/timer.py]: This last sentence is after the warning upstream (and in DriverStation which it is intended to match).

Comment (ariovistus) [reply wpilib/wpilib/timer.py]: fixed

**Pull request merged**